compiler/AST/symbol.cpp

```diff
@@ -303,6 +308,82 @@ bool VarSymbol::isParameter() const {
   return hasFlag(FLAG_PARAM) || immediate;
 }
 
+
+std::string VarSymbol::docsDirective() {
+  std::string result;
+  if (fDocsTextOnly) {
+    result = "";
+  } else {
+    // TODO: If this is a type (i.e. this->hasFlag(FLAG_TYPE_VARIABLE)), use
+    //       ".. type:: " as directive. (thomasvandoren, 2015-02-21)
+    if (this->isField) {
+      result = ".. attribute:: ";
+    } else {
+      result = ".. data:: ";
+    }
+  }
+  return this->hasFlag(FLAG_CONFIG) ? result + "config " : result;
+}
+
+
+void VarSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC) || this->hasFlag(FLAG_SUPER_CLASS)) {
+      return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  if (this->hasFlag(FLAG_TYPE_VARIABLE)) {
+    *file << "type ";
+  } else if (this->isConstant()) {
+    *file << "const ";
+  } else if (this->isParameter()) {
+    *file << "param ";
+  } else {
+    *file << "var ";
+  }
+
+  *file << this->name;
+
+  if (this->defPoint->exprType != NULL) {
+    *file << ": ";
+    this->defPoint->exprType->prettyPrint(file);
+  }
+
+  // FIXME: Do we want this for non-class globals/configs?
+  //        (thomasvandoren, 2015-02-21)
```

Comment:
I think a type field and a global type have sufficiently different uses to justify including the default value for fields but not for global type declarations.

Are there any types we cannot write?  I thought that was defined by default on all types.  In regard to the other two points - "can I pass it somewhere that an int is required" definitely feels like relying on implementation details for convenience (which should not be encouraged on principle) and "can I cast it to _" can be easily denoted by providing a cast function or a toInt() method.  In either case, failing to say the default value of a type does not prevent the user from relying on implementation details that they discover, it merely serves to make it clear that they shouldn't rely on these implementation details.  For instance, what if we wanted age to now be a class that stored information in addition to a numeral, such as an enum to choose between "days", "months" or "years" (and methods to output the correct number depending on the circumstances)?  In that case, we would not be able to provide such functionality without breaking someone's code and rightly so - we told them it was an int, now it isn't.  Whereas, if the underlying type of age was unknown when it was first introduced, we could switch and say "look, here are these new things you can do with age!" with very few people being the wiser (and only those people that rely on undocumented "features").  Sure, we can get away with changing the type under people now from the argument of "probably no one is relying on it", but as we get to be a more mature language we won't be able to do such things and would have to break the habit anyways.